### PR TITLE
Remove support for multi-dist compilation (BREAKING)

### PR DESCRIPTION
**Pull request by bajtos**
- Simplify `lb-tsc` to always use `outDir` and `target` as configured via `tsconfig.json` or CLI arguments.
- Simplify `lb-clean` to always require list of file globs to delete.
- Simplify `lb-mocha`, remove code converting `DIST` to the outDir based on compilation target.
- Remove already deprecated `lb-dist` command.

**BREAKING CHANGE**

We are no longer choosing outDir for you, you have to specify it explicitly. It is no longer possible to specify compilation target via non-option argument like `lb-tsc es2017`.

**Migration guide**

- Modify your `tsconfig.json` file and configure `dist` via `compilerOptions.outDir`

- If you are using target different from `es2017`, then configure it via  `compilerOptions.target`.

- Remove `es2017` and `--outDir dist` from lb-tsc arguments.

- Ensure that the output directory is listed in `lb-clean` arguments,   e.g. call `lb-clean dist`.

- When calling `lb-mocha`, replace `DIST` with the actual outDir value,  typically `dist`.

## Checklist

👉 [Read and sign the CLA (Contributor License Agreement)](https://cla.strongloop.com/agreements/strongloop/loopback-next) 👈

- [x] `npm test` passes on your machine
- [x] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- API Documentation in code was updated
- Documentation in [/docs/site](../tree/master/docs/site) was updated
- [x] Affected artifact templates in `packages/cli` were updated
- [x] Affected example projects in `examples/*` were updated

👉 [Check out how to submit a PR](https://loopback.io/doc/en/lb4/submitting_a_pr.html) 👈

### CODE REVIEW

**Comment by bajtos** [comment]
Note for reviewers: I organized the changes into a series of small(ish) commits to make the review easier. You may want to review the changes commit-by-commit.

**Comment by bajtos** [comment]
I'll have to split the last commit into two parts, so that only `@loopback/build` is considered to contain breaking changes.

**Comment by bajtos** [comment]
> I'll have to split the last commit into two parts, so that only `@loopback/build` is considered to contain breaking changes.

Done. 

The tests are failing on Node.js 8, I am able to reproduce the failure locally and will investigate.

File: packages/context/src/context.ts [hunk 1]
```diff
@@ -44,12 +50,6 @@ if (!Symbol.asyncIterator) {
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
   (Symbol as any).asyncIterator = Symbol.for('Symbol.asyncIterator');
 }
-/**
- * WARNING: This following import must happen after the polyfill. The
- * `auto-import` by an IDE such as VSCode may move the import before the
- * polyfill. It must be then fixed manually.
- */
-import {iterator, multiple} from 'p-event';
```
Comment (raymondfeng) [inline packages/context/src/context.ts]: Please revert this change as it breaks Node 8.

**Comment by raymondfeng** [comment]
@bajtos I fixed the `context.ts` issue.

**Comment by bajtos** [comment]
> I fixed the `context.ts` issue.

Thank you for the help!
